pdl-compiler/src/backends/rust/serializer.rs

```diff
@@ -373,6 +373,50 @@ impl Encoder {
                     shift,
                 });
             }
+            ast::FieldDesc::ElementSize { field_id, width, .. } => {
+                let field_name = field_id.to_ident();
+                let field_type = types::Integer::new(*width);
+                let field_element_size_name = format_ident!("{field_id}_element_size");
+                let packet_name = &self.packet_name;
+                self.tokens.extend(quote! {
+                    let #field_element_size_name = self.#field_name
+                        .get(0)
+                        .map_or(0, Packet::encoded_len);
+
+                    for (element_index, element) in self.#field_name.iter().enumerate() {
+                        if element.encoded_len() != #field_element_size_name {
+                            return Err(EncodeError::InvalidArrayElementSize {
+                                packet: #packet_name,
+                                field: #field_id,
+                                size: element.encoded_len(),
+                                expected_size: #field_element_size_name,
+                                element_index,
+                            })
+                        }
+                    }
+                });
+                if field_type.width > *width {
```

Comment:
The check should be present so long as the type is smaller than usize `(width < usize::BITS)`.